source/_docs/guides/jenkins.md

```diff
@@ -0,0 +1,310 @@
+
+1. Log into the Jenkins dashboard as an admin user. Click on **New Item**.
+
+    ![Jenkins dashboard](/source/docs/assets/images/integrations/jenkins_dash.png)
+
+2. Give the project a name with no spaces. Select **Freestyle Project** and click **OK** to save.
+
+    ![New Jenkins project](/source/docs/assets/images/integrations/new_job.png)
+
+3. The next page lets you configure options for this project. In the **General** tab, select "GitHub project" and enter the repository URL (e.g. `https://github.com/YOUR-ORG/YOUR-PROJECT`).
+
+4. In the **Source Code Management** tab select **Git**, then **Advanced**. Add the following information:
+
+    - **Repository URL**: The path to the .git file of your repository, e.g. `https://github.com/YOUR-ORG/YOUR-PROJECT.git`
+    - **Name**: `origin`
+    - **Refspec**: blank
+    - **Branch specifier**: `origin/*`
+    - **Additional Behaviours**: Add **Prune stale remote-tracking branches**:
+
+    ![SCM view](/source/docs/assets/images/integrations/scm_settings.png)
+
+5. We want code changes to trigger our build (as opposed to setting up a periodic build, for example). Under **Build Triggers**, Check the box labelled, "GitHub hook trigger for GITScm polling".
+
+6. Under **Build Environment**, check the box labelled "Inject environment variables to the build process."
+
+
+7. In the **Properties File Path** field, enter the path to your Jenkins environment variables file. On Debian-based systems, it's usually `/var/lib/jenkins/envVars.properties`. If it doesn't already exist, create an empty file.
+
+
+8. In the **Properties Content** field, add the following variable one per line, with no quotation marks:
+
+    - **SITE_ID=**your-site-name
+
+    ![Env vars view](/source/docs/assets/images/integrations/env_vars.png)
+
+9. Check "Use secret text(s) or file(s)" option. A new button called **Bindings** will appear. add a secret text binding. Name it **TERMINUS_TOKEN** and select the Terminus secret text credential from the dropdown:
+
+    ![Bindings](/source/docs/assets/images/integrations/jenkins-binding.png)
+
+### Add Build Steps
+
+Under the **Build** tab is a button labeled **Add build step**. These tasks will execute in sequence, and the job will quit if any fail. Add these in separate build steps, selecting **Execute shell** for all but steps 1 and 6.
+
+1. Set build status to "pending" on GitHub commit
+
+2. Jenkins logs into Pantheon:
+
+    ```nohighlight
+    #!/bin/bash
+    echo "Logging into Terminus"
+    terminus auth:login --machine-token=${TERMINUS_TOKEN}
+    ```
+
+3. Verifies the dev site is awake and in git mode. Note that these are separate build steps:
+
+    ```nohighlight
+    echo "Waking Dev environment."
+    terminus env:wake -n ${SITE_ID}.dev
+    ```
+
+    ```nohighlight
+    echo "Setting site to git mode."
+    terminus connection:set ${SITE_ID}.dev git
+    ```
+
+4. Jenkins creates a Multidev and pushes the new code to this environment
```

Comment:
Does this mean that every commit, even those to master, get put on a multidev first?